src/App.tsx

```diff
@@ -1,26 +1,205 @@
+import { filterTodosByComplated } from './utils/filterTodosByCompleted';
+import { Errors } from './types/Errors';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [completedCategory, setCompletedCategory] =
+    useState<TodoCompletedCategory>(TodoCompletedCategory.all);
+  const [errorMessage, setErrorMessage] = useState<Errors>(Errors.noneError);
+  const [updatedTodosId, setUpdatedTodosId] = useState([0]);
+
+  const filtredTodos = filterTodosByComplated(todos, completedCategory);
+  const countOfCompletedTodos = todos.filter(todo => todo.completed).length;
+  const countOfNotCompletedTodos = todos.length - countOfCompletedTodos;
+  const isSomeTodoComplated = todos.some(todo => todo.completed);
+
+  async function fetchTodosFromApi() {
+    try {
+      const todosFromApi = await getTodos();
+
+      setTodos(todosFromApi);
+    } catch {
+      setErrorMessage(Errors.loadError);
+    }
   }
 
+  async function fetchAddTodo(newTitle: string) {
+    let newTodo: Todo = {
+      id: 0,
+      title: newTitle,
+      userId: 0,
+      completed: false,
+    };
+
+    setTempTodo(newTodo);
+
+    try {
+      newTodo = await createTodo(newTitle);
+    } catch {
+      setErrorMessage(Errors.addError);
+    } finally {
+      setTempTodo(null);
+      if (newTodo.id !== 0) {
+        setTodos(currentTodos => [...currentTodos, newTodo]);
+
+        return true;
+      } else {
+        return false;
+      }
+    }
+  }
+
+  async function fetchDeleteTodo(todoId: number) {
+    try {
+      setUpdatedTodosId(current => [...current, todoId]);
+      await deleteTodo(todoId);
+      setTodos(currentTodos => currentTodos.filter(todo => todo.id !== todoId));
+    } catch {
+      setErrorMessage(Errors.deleteError);
+    } finally {
+      setUpdatedTodosId(current => current.filter(id => id !== todoId));
+    }
+  }
+
+  function deleteAllCompletedTodos() {
+    const completedTodosId: number[] = todos
+      .filter(todo => todo.completed)
+      .map(todo => todo.id);
+
+    Promise.allSettled(
+      completedTodosId.map(async id => {
+        await fetchDeleteTodo(id);
+      }),
+    );
+  }
+
+  async function fetchUpdateTodoCompleted(
+    todoId: number,
+    isCompleted: boolean,
+  ) {
+    try {
+      setUpdatedTodosId(current => [...current, todoId]);
+      await updateTodoCompleted(todoId, isCompleted);
+      setTodos(currentTodos =>
+        currentTodos.map(todo => {
+          if (todo.id === todoId) {
+            return { ...todo, completed: isCompleted };
+          }
+
+          return todo;
+        }),
+      );
+    } catch {
+      setErrorMessage(Errors.updateError);
+    } finally {
+      setUpdatedTodosId(current => current.filter(id => id !== todoId));
+    }
+  }
+
+  async function UpdateAllTodosCompleted() {
```

Comment:
Rename `UpdateAllTodosCompleted `to `updateAllTodosCompletion`
This makes it clear that the function is intended to update the completion status of all todos.